### PR TITLE
Checker and Error Classification

### PR DESCRIPTION
**Pull request by kanak8278**
Modified the Checker to check for transition loop.
Check for using Variables only if declared earlier.
The errors are classified into 2 categories: "errors", "warnings".

### CODE REVIEW

File: nl2dsl/__init__.py [hunk 1]
```diff
@@ -140,7 +140,7 @@ def _update_dsl_using_plan(self) -> Dict[str, List[Dict]]:
         return self.dsl
 
     def validate_dsl(self):
-        checker = Checker(self.dsl["dsl"], self.dsl["variables"])
+        checker = Checker(self.dsl["dsl"], self.dsl["variables"], self.dsl["config_vars"])
```
Comment (sameersegal) [inline nl2dsl/__init__.py]: Why doesn't the checker take the whole DSL

File: nl2dsl/utils/checker.py [hunk 2]
```diff
@@ -64,112 +250,80 @@ def variable_checker(self):
...
-                    self.errors.append(f"{key} transitions is missing")
+                    self.errors["errors"].append(f"{state_name} transitions is missing")
 
         return self.errors
 
 
 if __name__ == "__main__":
-    with open("bandhu/step1/gold.json") as f:
+
+    with open("bandhu/gold.json") as f:
```
Comment (sameersegal) [inline nl2dsl/utils/checker.py]: Remove this please

**Review by sameersegal — CHANGES_REQUESTED** [review]

**Comment by sameersegal** [comment]
The checker in this repo should be focused on specific DSL properties and ensuring that the DSL conforms to specification.

All other checks that are JB Specific need to move to the JB Engine repo

**Comment by kanak8278** [comment]
> The checker in this repo should be focused on specific DSL properties and ensuring that the DSL conforms to specification.
> 
> All other checks that are JB Specific need to move to the JB Engine repo

I am creating new issue for splitting the checker between Jugalbandi Studio and PwR-NL2DSL.
We can merge this for now

**Review by sameersegal — APPROVED** [review]

**Pull request merged**